packages/ComponentLibrary/src/components/PrimaryTab/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react';
+import {
+  Tabs,
+  Tab,
+  Box,
+  Menu,
+  MenuItem,
+  ListItemIcon,
+  Tooltip,
+} from '@mui/material';
+import CheckIcon from '@mui/icons-material/Check';
+import { PrimaryTabsProps, TabItem } from './types';
+import { menuStyle, styles, sx, tabIndicatorProps } from './styles';
+import IconButton from '../IconButton';
+
+const PrimaryTabs: React.FC<PrimaryTabsProps> = ({ tabs, onChange, icon }) => {
+  const [selectedTab, setSelectedTab] = useState(tabs[0]?.id || '');
+  const [hoveredTab, setHoveredTab] = useState<string | null>(null);
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+  const handleChange = (_: React.SyntheticEvent, newValue: string) => {
+    setSelectedTab(newValue);
+    if (onChange) {
+      onChange(newValue);
+    }
+  };
+
+  const handleMenuOpen = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleMenuClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleMenuItemClick = (id: string) => {
+    setSelectedTab(id);
+    if (onChange) {
+      onChange(id);
+    }
+  };
+
+  return (
+    <Box style={styles.containerBox}>
+      <Box style={styles.tabsContainer}>
+        <Tabs
+          value={selectedTab}
+          onChange={handleChange}
+          variant="scrollable"
+          scrollButtons="auto"
+          TabIndicatorProps={tabIndicatorProps}
+          aria-label="primary tabs"
+          sx={sx.tabs}>
+          {tabs.map((tab: TabItem) => {
+            const isSelected = selectedTab === tab.id;
+            const isHovered = hoveredTab === tab.id;
+
+            return (
+              <Tab
+                key={tab.id}
+                value={tab.id}
+                icon={
+                  tab.showInTab !== 'label' && tab.icon
+                    ? React.cloneElement(tab.icon as React.ReactElement, {
```

Comment:
In order to avoid importing the whole React module here, I suggest to import the cloneElement function:
`import { cloneElement } from 'react';`
About the types used, like React.ReactElement, you don't need to import it anymore, since the "React" namespace is global and you can reference types there without explicit import